python/lsst/analysis/drp/skyPlot.py

```diff
@@ -294,41 +329,78 @@ def skyPlot(self, catPlot, plotInfo, sumStats):
-            height = corners[2][1].asDegrees() - dec
-            patches.append(Rectangle(xy, width, height, alpha=0.3))
-            ras = [ra.asDegrees() for (ra, dec) in corners]
-            decs = [dec.asDegrees() for (ra, dec) in corners]
-            ax.plot(ras + [ras[0]], decs + [decs[0]], "k", lw=0.5)
-            cenX = ra + width / 2
-            cenY = dec + height / 2
-            if dataId != "tract":
-                ax.annotate(dataId, (cenX, cenY), color="k", fontsize=7, ha="center", va="center",
-                            path_effects=[pathEffects.withStroke(linewidth=2, foreground="w")])
+        if self.config.plotOutlines:
+            patches = []
+            for dataId in sumStats.keys():
+                (corners, stat) = sumStats[dataId]
+                ra = corners[0][0].asDegrees()
+                dec = corners[0][1].asDegrees()
+                xy = (ra, dec)
+                width = corners[2][0].asDegrees() - ra
+                height = corners[2][1].asDegrees() - dec
+                patches.append(Rectangle(xy, width, height, alpha=0.3))
+                ras = [ra.asDegrees() for (ra, dec) in corners]
+                decs = [dec.asDegrees() for (ra, dec) in corners]
+                ax.plot(ras + [ras[0]], decs + [decs[0]], "k", lw=0.5)
+                cenX = ra + width / 2
+                cenY = dec + height / 2
+                if dataId == "tract":
+                    minRa = np.min(ras)
+                    minDec = np.min(decs)
+                    maxRa = np.max(ras)
+                    maxDec = np.max(decs)
+                if dataId != "tract":
+                    ax.annotate(dataId, (cenX, cenY), color="k", fontsize=5, ha="center", va="center",
+                                path_effects=[pathEffects.withStroke(linewidth=2, foreground="w")])
 
         for (i, (xs, ys, colorVals, cmap, label)) in enumerate(toPlotList):
+            if "tract" not in sumStats.keys() or not self.config.plotOutlines:
+                minRa = np.min(xs)
+                maxRa = np.max(xs)
+                minDec = np.min(ys)
+                maxDec = np.max(ys)
             med = np.median(colorVals)
             mad = sigmaMad(colorVals)
-            vmin = med - 3*mad
-            vmax = med + 3*mad
-            scatterOut = ax.scatter(xs, ys, c=colorVals, cmap=cmap, s=4.0, vmin=vmin, vmax=vmax)
+            vmin = med - 2*mad
+            vmax = med + 2*mad
+            if self.config.fixAroundZero:
+                scaleEnd = np.max([np.fabs(vmin), np.fabs(vmax)])
+                vmin = -1*scaleEnd
+                vmax = scaleEnd
+            nBins = 50
+            xBins = np.linspace(minRa, maxRa, nBins)
+            yBins = np.linspace(minDec, maxDec, nBins)
+            binnedStats, xEdges, yEdges, binNums = binned_statistic_2d(xs, ys, colorVals, statistic="median",
+                                                                       bins=(xBins, yBins))
```

Comment:
You're not setting `xBins` and `yBins` here, rather, you're setting bin edges. I've modified this section a little, which makes that clearer, and also improves the legibility of HSC plots by aligning bins with patches:
```
            nBins = 45
            xBinEdges = np.linspace(minRa, maxRa, nBins+1)
            yBinEdges = np.linspace(minDec, maxDec, nBins+1)
            binnedStats, xEdges, yEdges, binNums = binned_statistic_2d(xs, ys, colorVals, statistic="median",
                                                                       bins=(xBinEdges, yBinEdges))
```

Reply:
The corner tract/patch plot colour bar labels in the `scatterTwoHistPlot`'s overlap with the main body figure. They need to be adjusted.

Reply:
For consistency, it would be nice if the n_galaxies and n_stars number counts in the `colorColorPlot` were printed in the same shaded box format as in the `skyPlot` plots.